doc/release/upcoming_changes/17492.deprecation.rst

```diff
@@ -0,0 +1,8 @@
+Inexact matches for `numpy.convolve` and `numpy.correlate` are deprecated
+-------------------------------------------------------------------------
+
+`numpy.convolve` and `numpy.correlate` now parses its ``mode`` argument in C
+instead of Python, and raises a warning when there are case insensitive and/or
+inexact matches found for ``mode`` argument in the functions.
+Pass full ``"same"``, ``"valid"``, ``"full"`` strings instead of
+``"s"``, ``"v"``, ``"f"`` for the ``mode`` argument.
```

Comment:
The fact it's in C is not release-note worthy, but the rest looks good:
```suggestion
`numpy.convolve` and `numpy.correlate` now emits a warning when there are case
insensitive and/or inexact matches found for ``mode`` argument in the functions.
Pass full ``"same"``, ``"valid"``, ``"full"`` strings instead of
``"s"``, ``"v"``, ``"f"`` for the ``mode`` argument.
```